### ISSUE TITLE
License Validation Fails for Non-English Languages

### ISSUE THREAD

**Issue by adamfisher** [adamfisher/Xamarin.Forms.VideoPlayer#40]
We are currently developing a Xamarin.Forms app that uses the Video Player component. Late in the game, we have encountered an issue with the license key that we don't quite understand and would like to get your input on.

When an Android Phone is set to use Danish Language, the license key check at startup crashes.
When the phone is switched to English, everything works as expected.

We have reproduced this problem on several phones. Please note, that you have to clear app data in settings between changes to the language, as otherwise it seems the registration check is not re-done.

The call stack at the crash is as follows:

```
    E  FATAL EXCEPTION: main
    E  Process: dk.tekniq.app, PID: 13338
    E  java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
    E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
    E  Caused by: java.lang.reflect.InvocationTargetException
    E      at java.lang.reflect.Method.invoke(Native Method)
    E      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
    E      ... 1 more
    E  Caused by: android.runtime.JavaProxyThrowable: Octane.Xam.VideoPlayer.Licensing.VideoPlayerLicenseException: The specified license key is invalid or incompatible with this mobile application.
    E  You are attempting to use functionality that is not included in the trial version of this component. Please consider purchasing a licensed copy at https://components.xamarin.com/view/video-player.
    E    at Octane.Xam.VideoPlayer.Licensing.VideoPlayerLicense.Initialize (System.String licenseKey) [0x000c3] in <e2fe4236b8e544b7b9a078bbbe49f92c>:0
    E    at Octane.Xam.VideoPlayer.Android.FormsVideoPlayer.Init (System.String licenseKey) [0x00025] in <552c34f1641948a989334ec218d29fdc>:0
    E    at Elvis.Droid.MainActivity.OnCreate (Android.OS.Bundle bundle) [0x0003f] in <b197e884ef9e4e45b12db7713b730d26>:0
    E    at Android.Support.V4.App.FragmentActivity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <7e083afc7c9045f59e01d9c14931060d>:0
    E    at (wrapper dynamic-method) System.Object:14a7e844-c05e-4b9e-98cc-e97db331890f (intptr,intptr,intptr)
    E      at md570fd9fefcf553cb649ae07dc125e2611.MainActivity.n_onCreate(Native Method)
    E      at md570fd9fefcf553cb649ae07dc125e2611.MainActivity.onCreate(MainActivity.java:29)
    E      at android.app.Activity.performCreate(Activity.java:6682)
    E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
    E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2619)
    E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
    E      at android.app.ActivityThread.-wrap12(ActivityThread.java)
    E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1478)
    E      at android.os.Handler.dispatchMessage(Handler.java:102)
    E      at android.os.Looper.loop(Looper.java:154)
    E      at android.app.ActivityThread.main(ActivityThread.java:6121)
    E      ... 3 more

       Process dk.tekniq.app (PID: 13338) ended

We have of obviously checked that we are initializing the video player as documented, but just in case, you can see the code below:

    public class MainActivity : FormsAppCompatActivity
    {
        const string VIDEO_KEY = "XXXXXXXXXX";

        protected override void OnCreate(Bundle bundle)
        {
            <...>
            Xamarin.Forms.Forms.Init(this, bundle);
            
            // Initiliaze video player
            FormsVideoPlayer.Init(VIDEO_KEY);
            <...>
        }
    }
```

Also, the fact that everything works as expected when the phone's language is English seems to indicate that the problem is not caused by the initialization code, or because we're not using an incorrect application id (which is also clear from the call stack).

**Comment by adamfisher** [comment]
In the interim, you should be able to temporarily set the license check to English and then set it back to the currently defined culture as a workaround:

```
var defaultCulture = Thread.CurrentThread.CurrentCulture;
Thread.CurrentThread.CurrentCulture = New CultureInfo("en-US");
FormsVideoPlayer.Init(...);
Thread.CurrentThread.CurrentCulture = defaultCulture;
```

**Comment by adamfisher** [comment]
This is addressed in release 2.0.0.

https://github.com/adamfisher/Xamarin.Forms.VideoPlayer/releases/tag/2.0.0
